src/code/z_demo.c

```diff
@@ -264,107 +296,131 @@ void func_80064824(PlayState* play, CutsceneContext* csCtx, CsCmdBase* cmd) {
+            if (isFirstFrame) {
                 Play_SetViewpoint(play, VIEWPOINT_LOCKED);
             }
             break;
-        case 15:
-            if (sp3F != 0) {
+
+        case CS_MISC_SHOW_TITLE_CARD:
+            if (isFirstFrame) {
                 TitleCard_InitPlaceName(play, &play->actorCtx.titleCtx, player->giObjectSegment, 160, 120, 144, 24, 20);
             }
             break;
-        case 16:
-            if (sp3F != 0) {
+
+        case CS_MISC_QUAKE_START:
+            if (isFirstFrame) {
                 sQuakeIndex = Quake_Request(GET_ACTIVE_CAM(play), QUAKE_TYPE_6);
                 Quake_SetSpeed(sQuakeIndex, 0x7FFF);
                 Quake_SetPerturbations(sQuakeIndex, 4, 0, 1000, 0);
                 Quake_SetDuration(sQuakeIndex, 800);
             }
             break;
-        case 17:
-            if (sp3F != 0) {
+
+        case CS_MISC_QUAKE_STOP:
+            if (isFirstFrame) {
                 Quake_RemoveRequest(sQuakeIndex);
             }
             break;
-        case 18:
+
+        case CS_MISC_STOP_STORM_AND_ADVANCE_TO_DAY:
             play->envCtx.precipitation[PRECIP_RAIN_MAX] = 0;
             play->envCtx.stormRequest = STORM_REQUEST_STOP;
+
             if (gSaveContext.dayTime < CLOCK_TIME(7, 0)) {
                 gSaveContext.dayTime += 30;
             }
+
             if (play->envCtx.precipitation[PRECIP_RAIN_CUR] == 0) {
                 gWeatherMode = WEATHER_MODE_CLEAR;
                 Audio_SetNatureAmbienceChannelIO(NATURE_CHANNEL_RAIN, CHANNEL_IO_PORT_1, 0);
             }
             break;
-        case 19:
+
+        case CS_MISC_SET_FLAG_FAST_WINDMILL:
             SET_EVENTCHKINF(EVENTCHKINF_65);
             break;
-        case 20:
+
+        case CS_MISC_SET_FLAG_WELL_DRAINED:
             SET_EVENTCHKINF(EVENTCHKINF_67);
             break;
-        case 21:
+
+        case CS_MISC_SET_FLAG_LAKE_HYLIA_RESTORED:
             SET_EVENTCHKINF(EVENTCHKINF_69);
             break;
-        case 22:
-            D_801614B0.r = 255;
-            D_801614B0.g = 255;
-            D_801614B0.b = 255;
-            D_801614B0.a = 255;
+
+        case CS_MISC_VISMONO_BLACK_AND_WHITE:
+            gVisMonoColor.r = 255;
+            gVisMonoColor.g = 255;
+            gVisMonoColor.b = 255;
+            gVisMonoColor.a = 255;
             break;
-        case 23:
-            D_801614B0.r = 255;
-            D_801614B0.g = 180;
-            D_801614B0.b = 100;
-            D_801614B0.a = 255.0f * temp;
+
+        case CS_MISC_VISMONO_SEPIA:
+            gVisMonoColor.r = 255;
+            gVisMonoColor.g = 180;
+            gVisMonoColor.b = 100;
+            gVisMonoColor.a = 255.0f * lerp;
             break;
-        case 24:
+
+        case CS_MISC_HIDE_ROOM:
             play->roomCtx.curRoom.segment = NULL;
             break;
-        case 25:
+
+        case CS_MISC_TIME_ADVANCE_TO_NIGHT:
             gSaveContext.dayTime += 30;
```

Comment:
Interestingly, this matches:
```suggestion
            gSaveContext.dayTime += CLOCK_TIME(0, 40 / 60.0f);
```
(+= 40 seconds)
but :shrug:

Reply:
i think i prefer the raw value tbh